testsuite/shared/src/main/java/org/jboss/as/test/integration/security/common/CoreUtils.java

```diff
@@ -631,4 +631,19 @@ public static String encodeQueryParam(final String paramName, final String param
         }
         return response;
     }
+
+
+    /**
+     * Creates a temporary folder name with given name prefix.
+     *
+     * @param prefix folder name prefix
+     * @return created folder
+     */
+    public static File createTemporaryFolder(String prefix) throws IOException {
+        File file = File.createTempFile(prefix, "", null);
+        LOGGER.debugv("Creating temporary folder {0}", file);
+        file.delete();
+        file.mkdir();
+        return file;
```

Comment:
You can achieve the same thing here with `Files.createTempDirectory()` then there is no need to do all the extra deletes and creations.